crates/accelerate/src/gate_direction.rs

```diff
@@ -142,8 +157,505 @@ where
+        &vec![&q1, &q0],
+    )?;
+
+    Ok(new_dag.clone())
+}
+
+fn swap_replacement_dag(py: Python) -> PyResult<DAGCircuit> {
+    let qreg = create_qreg(py, 2)?;
+    let new_dag = &mut DAGCircuit::new(py)?;
+    new_dag.add_qreg(py, &qreg)?;
+
+    let (q0, q1) = (qreg_bit(py, &qreg, 0)?, qreg_bit(py, &qreg, 1)?);
+
+    apply_op_back(
+        py,
+        new_dag,
+        &std_gate(py, StandardGate::SwapGate)?,
+        &vec![&q1, &q0],
+    )?;
+
+    Ok(new_dag.clone())
+}
+
+fn rxx_replacement_dag(py: Python, param: &[Param]) -> PyResult<DAGCircuit> {
+    let qreg = create_qreg(py, 2)?;
+    let new_dag = &mut DAGCircuit::new(py)?;
+    new_dag.add_qreg(py, &qreg)?;
+
+    let (q0, q1) = (qreg_bit(py, &qreg, 0)?, qreg_bit(py, &qreg, 1)?);
+
+    apply_op_back(
+        py,
+        new_dag,
+        &parameterized_std_gate(py, StandardGate::RXXGate, param)?,
+        &vec![&q1, &q0],
+    )?;
+
+    Ok(new_dag.clone())
+}
+
+fn ryy_replacement_dag(py: Python, param: &[Param]) -> PyResult<DAGCircuit> {
+    let qreg = create_qreg(py, 2)?;
+    let new_dag = &mut DAGCircuit::new(py)?;
+    new_dag.add_qreg(py, &qreg)?;
+
+    let (q0, q1) = (qreg_bit(py, &qreg, 0)?, qreg_bit(py, &qreg, 1)?);
+
+    apply_op_back(
+        py,
+        new_dag,
+        &parameterized_std_gate(py, StandardGate::RYYGate, param)?,
+        &vec![&q1, &q0],
+    )?;
+
+    Ok(new_dag.clone())
+}
+
+fn rzz_replacement_dag(py: Python, param: &[Param]) -> PyResult<DAGCircuit> {
+    let qreg = create_qreg(py, 2)?;
+    let new_dag = &mut DAGCircuit::new(py)?;
+    new_dag.add_qreg(py, &qreg)?;
+
+    let (q0, q1) = (qreg_bit(py, &qreg, 0)?, qreg_bit(py, &qreg, 1)?);
+
+    apply_op_back(
+        py,
+        new_dag,
+        &parameterized_std_gate(py, StandardGate::RZZGate, param)?,
+        &vec![&q1, &q0],
+    )?;
+
+    Ok(new_dag.clone())
+}
+
+fn rzx_replacement_dag(py: Python, param: &[Param]) -> PyResult<DAGCircuit> {
+    let qreg = create_qreg(py, 2)?;
+    let new_dag = &mut DAGCircuit::new(py)?;
+    new_dag.add_qreg(py, &qreg)?;
+
+    let (q0, q1) = (qreg_bit(py, &qreg, 0)?, qreg_bit(py, &qreg, 1)?);
+
+    apply_op_back(py, new_dag, &std_gate(py, StandardGate::HGate)?, &vec![&q0])?;
+    apply_op_back(py, new_dag, &std_gate(py, StandardGate::HGate)?, &vec![&q1])?;
+    apply_op_back(
+        py,
+        new_dag,
+        &parameterized_std_gate(py, StandardGate::RZXGate, param)?,
+        &vec![&q1, &q0],
+    )?;
+    apply_op_back(py, new_dag, &std_gate(py, StandardGate::HGate)?, &vec![&q0])?;
+    apply_op_back(py, new_dag, &std_gate(py, StandardGate::HGate)?, &vec![&q1])?;
+
+    Ok(new_dag.clone())
+}
+
+#[pymodule]
 pub fn gate_direction(m: &Bound<PyModule>) -> PyResult<()> {
     m.add_wrapped(wrap_pyfunction!(py_check_with_coupling_map))?;
     m.add_wrapped(wrap_pyfunction!(py_check_with_target))?;
+    m.add_wrapped(wrap_pyfunction!(py_fix_with_coupling_map))?;
+    m.add_wrapped(wrap_pyfunction!(py_fix_with_target))?;
```

Comment:
Is there a reason why we're callingn these `fix_...` and not `run_...`?

Reply:
As stated in https://github.com/Qiskit/qiskit/pull/13069#discussion_r1816967910 - the main reason being that both `GateDirection` and `CheckGateDirection` live in this file and there are the `check_...` couple and the `fix_...` couple, each pair corresponds to their respective passes. I could BTW just have one run method for each pass which could take both Target and a coupling map but I actually liked better the modularity of having 2 entry points (one for target and one for coupling map), where each defines a specialized closure but both call the same function for each pass. These are being called by `GateDirection.run()` anyway. That said, should we follow a convention that all pass entry functions should be named `run..,`? If so I'll prefix the four python functions in the file with it. Let me know.

Reply:
It is good to have conventions to keep everything consistent. That said, we haven't had any of them apply to transpiler passes in rust yet. We do have the main run method calling upon other helper methods, so having the rust methods that represent the main `run` method of the pass have `run` somewhere in the name would be nice. I did it as a suffix for #13237, but having it as a prefix sounds more elegant. Maybe we should open an issue about these conventions here (sort of what #12936 proposes).

Reply:
On a second thought and after considering some alternatives I actually liked most this one: `py_check_direction_coupling_map` (and similarly for the `fix` and `target` variants). I find it more informational and succinct than other options like: `py_run_check_with_coupling_map` or `py_run_check_coupling_map` or `py_run_check_direction_coupling_map` etc... Also the `py_` prefix in it indicates that it's the Python entry point, hence also the pass entry point so the `run` prefix is not really required for this purpose. So unless you feel strongly against it, I would prefer to stay with this more informational naming scheme here until we decide on a convention for the pass entry point names.